x-pack/plugins/global_search_bar/public/types.ts

```diff
@@ -6,8 +6,42 @@
  */
 
 import { UiCounterMetricType } from '@kbn/analytics';
+export { METRIC_TYPE } from '@kbn/analytics';
 
+/* @internal */
+export enum COUNT_METRIC {
+  UNHANDLED_ERROR = 'unhandled_error',
+  SHORTCUT_USED = 'shortcut_used',
+  SEARCH_BLUR = 'search_blur',
+  SEARCH_FOCUS = 'search_focus',
+  SEARCH_REQUEST = 'search_request',
+}
+
+/* @internal */
+export enum CLICK_METRIC {
+  USER_NAVIGATED_TO_APPLICATION = 'user_navigated_to_application',
+  USER_NAVIGATED_TO_SAVED_OBJECT = 'user_navigated_to_saved_object',
+}
+
+export interface TrackedApplicationClick {
+  application: string;
+}
+export interface TrackedSavedObjectClick {
+  type: string;
+}
+export interface TrackedError {
+  message: string | Error;
+}
+
+export type TrackingContext = (TrackedApplicationClick | TrackedSavedObjectClick | TrackedError) & {
+  searchValue?: string;
```

Comment:
TODO: add `searchRank` to track the position of the item selected for navigation